### PR TITLE
tests: fix several stability issues

### PR DESCRIPTION
**Pull request by bobotu**
### What problem does this PR solve?
Fixes https://github.com/pingcap/tidb/issues/18236 by upgrading unistore to https://github.com/ngaut/unistore/pull/404.

Fixes https://github.com/pingcap/tidb/issues/18235 by updating `interestingGoroutines`.

Stabilize `TestShardRowIDBitsStep` by increasing `shard_row_id_bits`.

### Release note 

-  No release note

### CODE REVIEW

File: util/testleak/leaktest.go [hunk 1]
```diff
@@ -31,38 +31,57 @@ import (
...
+		"runtime.MHeap_Scavenger",
+		"created by os/signal.init",
+		// these go routines are async terminated, so they may still alive after test end, thus cause
+		// false positive leak failures
+		"google.golang.org/grpc.(*addrConn).resetTransport",
+		"google.golang.org/grpc.(*ccBalancerWrapper).watcher",
+		"github.com/pingcap/goleveldb/leveldb/util.(*BufferPool).drain",
+		"github.com/pingcap/goleveldb/leveldb.(*DB).compactionError",
+		"github.com/pingcap/goleveldb/leveldb.(*DB).mpoolDrain",
+		"go.etcd.io/etcd/v3/pkg/logutil.(*MergeLogger).outputLoop",
```
Comment (bobotu) [inline util/testleak/leaktest.go:61]: I have found this and `created by os/signal.init` in other failure tests

They are also in etcd's ignore list
https://github.com/etcd-io/etcd/blob/master/pkg/testutil/leak.go#L121-L132

File: table/tables/tables_test.go [hunk 2]
```diff
@@ -400,15 +400,15 @@ func (ts *testSuite) TestShardRowIDBitsStep(c *C) {
 	tk := testkit.NewTestKit(c, ts.store)
 	tk.MustExec("use test")
 	tk.MustExec("drop table if exists shard_t;")
-	tk.MustExec("create table shard_t (a int) shard_row_id_bits = 7;")
+	tk.MustExec("create table shard_t (a int) shard_row_id_bits = 15;")
```
Comment (wjhuang2016) [inline table/tables/tables_test.go:403]: Why do you think that this test is unstable?

**Pull request merged**